### PR TITLE
Ayesha/add slds linter

### PR DESCRIPTION
**Pull request by ayeshakmaz**
### What does this PR do?
An SLDS Linter that produces a list of warnings about the usage of deprecated —BEM syntax in Lightning files, plus options to fix said warnings

### Tasks

- [x] Rebase onto latest develop 
- [x] Externalize strings (do a first pass and then we will have our doc writer review)
- [ ] Ensure that generated .vsix file is complete and can run on Mac, Linux, and Windows 
- [ ] Add tests
- [x] Add documentation (do a first quick write-up and then we will have our doc writer review)

### CODE REVIEW

File: packages/salesforcedx-slds-linter/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4740 @@
+{
```
Comment (vazexqi) [inline packages/salesforcedx-slds-linter/package-lock.json]: Could you remove this file? Let's not commit the package-lock.json files for now.

File: packages/salesforcedx-vscode-lightning/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as languageServer from '@salesforce/salesforcedx-slds-linter/out/src/client';
+import * as vscode from 'vscode';
+
+export function activate(context: vscode.ExtensionContext) {
+  console.log('SFDX SLDS Linter Extension Activated');
+  vscode.workspace
+    .findFiles('**/staticresources/*.resource', '**/node_modules/**')
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-lightning/src/index.ts]: Why are we looking into the node_modules folder? Do we have control over those?

File: packages/salesforcedx-vscode-lightning/package.json [hunk 3]
```diff
@@ -46,6 +47,11 @@
     "postinstall": "node ./node_modules/vscode/bin/install",
     "test": "node ./node_modules/vscode/bin/test"
   },
+  "activationEvents": [
+    "workspaceContains:sfdx-project.json",
+    "sfdx.debug.provideInitialConfigurations"
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-lightning/package.json]: Is there a reason why you need the "sfdx.debug.provideInitialConfiguration" as an activation event?

File: packages/salesforcedx-slds-linter/src/client/index.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            mutator.replace(
+              client.protocol2CodeConverter.asRange(edit.range),
+              edit.newText
+            );
+          }
+        })
+        .then(success => {
+          if (!success) {
+            window.showErrorMessage(
+              'Failed to apply SLDS Validator fixes to the document'
```
Comment (vazexqi) [inline packages/salesforcedx-slds-linter/src/client/index.ts]: Could you externalize this? See the examples in the other modules where we use the nls.... util.

File: packages/salesforcedx-slds-linter/src/server/index.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      const fixedString = found[0].replace('--', '_');
+      problems++;
+      const diagnostic = <Diagnostic>{
+        code: `0${fixedString}`,
+        severity: DiagnosticSeverity.Warning,
+        range: {
+          start: { line: i, character: index },
+          end: { line: i, character: index + foundStringLength }
+        },
+        message: `Deprecated SLDS class name (v2.3.1): ${line.substr(
```
Comment (vazexqi) [inline packages/salesforcedx-slds-linter/src/server/index.ts]: Can we externalize the message as well?

File: packages/salesforcedx-slds-linter/src/server/index.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    switch (code) {
+      case '0': {
+        edits.push({
+          range: diagnostic.range,
+          newText: replacementStr
+        });
+
+        result.push(
+          Command.create(
+            `Fix: ${diagnostic.message}`,
```
Comment (vazexqi) [inline packages/salesforcedx-slds-linter/src/server/index.ts]: Externalize the message?

File: packages/salesforcedx-slds-linter/src/server/index.ts [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+          default: {
+            codeMessage = 'same problems';
+          }
+        }
+
+      }
+    }
+    result.push(
+      Command.create(
+        `Fix: All ${codeMessage}`,
```
Comment (vazexqi) [inline packages/salesforcedx-slds-linter/src/server/index.ts]: Externalize?

**Comment by vazexqi** [comment]
Tasks:
 
- [ ] Rebase onto latest develop
- [ ] Externalize strings
- [ ] Ensure that generated .vsix file is complete and can run on Mac, Linux, and Windows
- [ ] Add tests
- [ ] Add documentation

File: packages/salesforcedx-vscode-lightning/src/index.ts (hunk 2 above)
Comment (ayeshakmaz) [reply packages/salesforcedx-vscode-lightning/src/index.ts]: The second arg is actually an exclude param, but looking at it now it doesn't make sense, I'll remove it

File: packages/salesforcedx-vscode-lightning/package.json (hunk 3 above)
Comment (ayeshakmaz) [reply packages/salesforcedx-vscode-lightning/package.json]: Nope, think it came from copying another package.json, will fix

**Comment by vazexqi** [comment]
Closing in favor of #101, which has been rebased more cleanly.

**Pull request closed without merging**